proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,454 @@
+
+This design retains Rust-inspired custom delimiters, offers all the features of both "cooked" and "raw" strings, introduces raw string interpolation, and does this _all_ without adding a new special-purpose string type to Swift.
+
+Yes, this approach requires work:
+
+* You must use pound signs for any raw string.
+* You must use a more cumbersome interpolation sequence for raw and cooked strings than conventional strings.
+
+Hopefully the tradeoffs are worth it in terms of added expressibility and the resulting design is sufficiently elegant to pass muster.
+
+### Updated Escape Character Sequences Reference
+
+Swift string literals may include the following special character sequences. Swift's escape delimiter begins with a backslash (Reverse Solidus, U+005C), and is followed by zero or more pound signs (Number Sign, U+0023). An escape delimiter in a string literal must match the number of pound signs used to delimit either end of the string.
+
+| Sequence | Escape Characters | Result |
+|----------|------------|--------|
+| Escape Delimiter + `0` | Digit Zero (U+0030)| Boundary Neutral / Null character U+0000 |
+| Escape Delimiter + `\` | Reverse Solidus (U+005C) | Punctuation / Backslash U+005C |
+| Escape Delimiter + `t` | Latin Small Letter T (U+0074) | Character Tabulation / Horizontal Tab U+0009 |
+| Escape Delimiter + `n` | Latin Small Letter N (U+006E) | Paragraph Separator / Line Feed U+000A |
+| Escape Delimiter + `r` | Latin Small Letter R (U+0072) | Paragraph Separator / Carriage Return U+000D |
+| Escape Delimiter + `"` | Quotation Mark (U+0022) | Punctuation / Quotation Mark U+0022 |
+| Escape Delimiter + `'` | Apostrophe (U+0027) | Punctuation / Apostrophe (Single Quote) U+0027 |
+| Escape Delimiter + `u{n}` | Latin Small Letter U (U+0055), Left Curly Bracket (U+007B), Right Curly Bracket (U+007D) | An arbitrary Unicode scalar, where *n* is a 1–8 digit hexadecimal number with a value equal to a valid Unicode code point |
+| Escape Delimiter + `(...)` | Left Parenthesis (U+0028), Right Parenthesis (U+0029) | An interpolated expression |
+
+
+## Examples
+
+Consider the text `\#1`:
+
+```
+// What you type today
+"\\#1" // escape delimiter + backslash + # + 1
+
+// What you type using the new system
+"\\#1" // the same
+
+// What you type with a single pound string delimiter
+#"\#\#1"# // escape delimiter + backslash + # + 1
+
+// Or you can adjust the string delimiter for a raw string:
+##"\#1"## // backslash + # + 1, no escape sequence
+```
+
+Adjusting string delimiters allows you to eliminate escape sequences to present text as intended for use:
+
+```
+#"c:\windows\system32"# // vs. "c:\\windows\\system32"
+#"\d{3) \d{3} \d{4}"# // vs "\\d{3) \\d{3} \\d{4}"
+
+#"a string with "double quotes" in it"#
+
+##"a string that needs "# in it"##
+
+#"""
+    a string with 
+    """
+    in it
+    """#
+```
+
+The following example terminates with backslash-r-backslash-n:
```

Comment:
Just for clarity, I would add `, not a carriage return and line feed` to the end.